config/src/main/java/org/springframework/security/config/annotation/method/configuration/Jsr250MethodSecurityConfiguration.java

```diff
@@ -40,28 +50,54 @@
 @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
 final class Jsr250MethodSecurityConfiguration {
 
-	private final Jsr250AuthorizationManager jsr250AuthorizationManager = new Jsr250AuthorizationManager();
+	private ObservationJsr250AuthorizationManager jsr250AuthorizationManager;
+
+	private AuthorizationManagerBeforeMethodInterceptor interceptor;
 
-	private SecurityContextHolderStrategy securityContextHolderStrategy = SecurityContextHolder
-			.getContextHolderStrategy();
+	@Autowired
+	Jsr250MethodSecurityConfiguration(ApplicationContext context) {
+		this.jsr250AuthorizationManager = new ObservationJsr250AuthorizationManager(context);
+		this.interceptor = AuthorizationManagerBeforeMethodInterceptor.jsr250(this.jsr250AuthorizationManager);
+	}
 
 	@Bean
 	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
 	Advisor jsr250AuthorizationMethodInterceptor() {
-		AuthorizationManagerBeforeMethodInterceptor interceptor = AuthorizationManagerBeforeMethodInterceptor
-				.jsr250(this.jsr250AuthorizationManager);
-		interceptor.setSecurityContextHolderStrategy(this.securityContextHolderStrategy);
-		return interceptor;
+		return this.interceptor;
 	}
 
 	@Autowired(required = false)
 	void setGrantedAuthorityDefaults(GrantedAuthorityDefaults grantedAuthorityDefaults) {
-		this.jsr250AuthorizationManager.setRolePrefix(grantedAuthorityDefaults.getRolePrefix());
+		this.jsr250AuthorizationManager.manager.setRolePrefix(grantedAuthorityDefaults.getRolePrefix());
 	}
 
 	@Autowired(required = false)
 	void setSecurityContextHolderStrategy(SecurityContextHolderStrategy securityContextHolderStrategy) {
-		this.securityContextHolderStrategy = securityContextHolderStrategy;
+		this.interceptor.setSecurityContextHolderStrategy(securityContextHolderStrategy);
+	}
+
+	private static final class ObservationJsr250AuthorizationManager implements AuthorizationManager<MethodInvocation> {
+
+		private Jsr250AuthorizationManager manager = new Jsr250AuthorizationManager();
```

Comment:
I don't want to reuse the class because I think that it is going to go away. Right now it's there because of a bean loading cycle. In a Boot application -- when this class uses `@Autowired` to get the `ObservationRegistry` -- the registry gets instantiated too early by Boot and Boot's `ObservationRegistry` auto-configuration breaks.